### PR TITLE
Fix #744  Fix #745 - fixup icon and Can't see QtWebEngine pages on macOS

### LINKED ISSUE #744: [v1.8.0-rc3] os.icns is outdated
Issue overview
--------------

image

## Current Behavior

image

## Possible Solution

Update the os.icns to match the new icon.

### Environment

Some additional details about your environment for this issue (if relevant):
 - Platform (Operating system, version): macOS
 - Version of OpenStudioApplication (if using an intermediate build, include SHA): v1.8.0-rc3

### Context

* #743

### LINKED ISSUE #745: [v1.8.0-rc3] Can't see QtWebEngine pages on macOS
Issue overview
--------------

## Current Behavior

Cannot see QtWebengine pages

image

## Steps to reproduce

```
OPENSTUDIO_APPLICATION_DEBUG=1 QT_DEBUG_PLUGINS=1 /Applications/OpenStudioApplication-1.8.0-rc3/OpenStudioApp.app/Contents/MacOS/OpenStudioApp
```

```
Debug: checking directory path "/Applications/OpenStudioApplication-1.8.0-rc3/OpenStudioApp.app/Contents/MacOS/tls" ... ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No TLS backend is available ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No TLS backend is available ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No TLS backend is available ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No TLS backend is available ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))

Warning: No functional TLS backend was found ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: QSslSocket::connectToHostEncrypted: TLS initialization failed ((null):0, (null))
Warning: QLayout: Attempting to add QLayout "" to openstudio::PreviewWebView "", which already has a layout ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: No functional TLS backend was found ((null):0, (null))
Warning: QSslSocket::connectToHostEncrypted: TLS initialization failed ((null):0, (null))
```

In chrome I see nothing

image

## Possible Solution

?

### Environment

Some additional details about your environment for this issue (if relevant):
 - Platform (Operating system, version): macOS, M1
 - Version of OpenStudioApplication (if using an intermediate build, include SHA): v1.8.0-rc3

### Context

* #743

### PR DESCRIPTION
**Pull request by jmarrec**
* Fix #744 
* Fix #745

### CODE REVIEW

File: icons/os.icns
Comment (jmarrec) [inline icons/os.icns:1]: image

Generated via 
```
iconutil -c icns os.iconset
```

File: CMake/QtWebEngineProcess.entitlements [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+    <key>com.apple.security.cs.disable-executable-page-protection</key>
```
Comment (jmarrec) [inline CMake/QtWebEngineProcess.entitlements:5]: entitlement for QtWebEngineProcess

File: src/openstudio_app/OpenStudioApp.rc.in [hunk 2]
```diff
@@ -15,7 +15,7 @@ BEGIN
             VALUE "FileDescription",  "OpenStudio Application for Energy Modeling\0"
             VALUE "FileVersion",      "${OPENSTUDIOAPPLICATION_VERSION}\0"
             VALUE "InternalName",     "OpenStudioApp\0"
-            VALUE "LegalCopyright",   "Copyright (c) 2020-2022, OpenStudio Coalition and other contributors. All rights reserved..\0"
+            VALUE "LegalCopyright",   "Copyright (c) 2020-${CURRENT_YEAR}, OpenStudio Coalition and other contributors. All rights reserved..\0"
```
Comment (jmarrec) [inline src/openstudio_app/OpenStudioApp.rc.in:18]: outdated, never again!

**Comment by jmarrec** [comment]
Ok, Icon is fixed, but still can't see the geometry editor...

image

File: src/openstudio_app/CMakeLists.txt [hunk 3]
```diff
@@ -547,4 +547,18 @@ endif()
...
+    CODE "
+    include(\"${CMAKE_CURRENT_FUNCTION_LIST_FILE}\")
+    codesign_files_macos(
+      FILES \"\${CMAKE_INSTALL_PREFIX}/$<TARGET_BUNDLE_DIR_NAME:${target_name}>/Contents/Frameworks/QtWebEngineCore.framework/Versions/A/Helpers/QtWebEngineProcess.app/Contents/MacOS/QtWebEngineProcess\"
+      SIGNING_IDENTITY \"${CPACK_CODESIGNING_DEVELOPPER_ID_APPLICATION}\"
+      IDENTIFIER \"${CPACK_CODESIGNING_MACOS_IDENTIFIER}.QtWebEngineProcess\"
+      ENTITLEMENTS \"${CMAKE_CURRENT_SOURCE_DIR}/../../cmake/QtWebEngineProcess.entitlements\"
+      FORCE VERBOSE
+      )
+  "
```
[the remark this answers is not in the document]
Comment (jmarrec) [reply src/openstudio_app/CMakeLists.txt:561]: Ok 

```
$ cat /Users/julien/Qt/6.5.2/macos/lib/QtWebEngineCore.framework/Versions/A/Helpers/QtWebEngineProcess.app/Contents/Resources/QtWebEngineProcess.entitlements
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>com.apple.security.cs.allow-unsigned-executable-memory</key>
	<true/>
	<key>com.apple.security.cs.disable-library-validation</key>
	<true/>
	<key>com.apple.security.cs.allow-jit</key>
	<true/>
        <key>com.apple.security.cs.disable-executable-page-protection</key>
        <true/>
</dict>
</plist>
```

https://doc.qt.io/qt-6/qtwebengine-deploying.html#macos-specific-deployment-steps

File: src/openstudio_app/CMakeLists.txt [hunk 4]
```diff
@@ -547,4 +547,18 @@ endif()
...
   register_install_codesign_target(${target_name} "." "OpenStudioApp")
+
+  install(
+    CODE "
+    include(\"${CMAKE_CURRENT_FUNCTION_LIST_FILE}\")
+    codesign_files_macos(
+      FILES \"\${CMAKE_INSTALL_PREFIX}/$<TARGET_BUNDLE_DIR_NAME:${target_name}>/Contents/Frameworks/QtWebEngineCore.framework/Versions/A/Helpers/QtWebEngineProcess.app/Contents/MacOS/QtWebEngineProcess\"
+      SIGNING_IDENTITY \"${CPACK_CODESIGNING_DEVELOPPER_ID_APPLICATION}\"
+      IDENTIFIER \"${CPACK_CODESIGNING_MACOS_IDENTIFIER}.QtWebEngineProcess\"
+      ENTITLEMENTS \"${QtWebEngineProcess}/Contents/Resources/QtWebEngineProcess.entitlements\"
```
Comment (jmarrec) [inline src/openstudio_app/CMakeLists.txt:558]: Grab the entitlements directly from the Qt install dir.

**Comment by macumber** [comment]
Awesome job @jmarrec !

**Pull request merged**